### PR TITLE
Fixes for application url missing when configuring with workflow and pagination issue with publisher and store rest apis

### PR DESCRIPTION
**Pull request by irham0019**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java [hunk 1]
```diff
@@ -550,6 +550,27 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+     */
+    public static APIListDTO fromAPIListToDTO(List<API> apiList) {
+        APIListDTO apiListDTO = new APIListDTO();
+        List<APIInfoDTO> apiInfoDTOs = apiListDTO.getList();
+        if (apiInfoDTOs == null) {
+            apiInfoDTOs = new ArrayList<>();
+            apiListDTO.setList(apiInfoDTOs);
+        }
+
+        for (int i = 0; i <= apiList.size() -1; i++) {
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: Shall we add a null validation for apiList before executing the loop ?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java [hunk 2]
```diff
@@ -210,6 +210,27 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+    /**
+     * Converts a List object of APIs into a DTO
+     *
+     * @param apiList List of APIs
+     * @return APIListDTO object containing APIDTOs
+     */
+    public static APIListDTO fromAPIListToDTO(List<API> apiList) {
+        APIListDTO apiListDTO = new APIListDTO();
+        List<APIInfoDTO> apiInfoDTOs = apiListDTO.getList();
+        if (apiInfoDTOs == null) {
```
Comment (isurulucky) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java]: The APIListDTO class creates a new APIInfoDTO list when you are creating a new object out of APIListDTO class. Hence we can remove this null check.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java [hunk 3]
```diff
@@ -210,6 +210,24 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+    /**
+     * Converts a List object of APIs into a DTO
+     *
+     * @param apiList List of APIs
+     * @return APIListDTO object containing APIDTOs
+     */
+    public static APIListDTO fromAPIListToDTO(List<API> apiList) {
+        APIListDTO apiListDTO = new APIListDTO();
+        List<APIInfoDTO> apiInfoDTOs = apiListDTO.getList();
+        if(apiList !=null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java [hunk 4]
```diff
@@ -550,6 +550,24 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+    /**
+     * Converts a List object of APIs into a DTO
+     *
+     * @param apiList List of APIs
+     * @return APIListDTO object containing APIDTOs
+     */
+    public static APIListDTO fromAPIListToDTO(List<API> apiList) {
+        APIListDTO apiListDTO = new APIListDTO();
+        List<APIInfoDTO> apiInfoDTOs = apiListDTO.getList();
+        if(apiList !=null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: Formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java [hunk 5]
```diff
@@ -210,6 +210,24 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+            for (int i = 0; i <= apiList.size() - 1; i++) {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java]: or you could simply say for (int i = 0; i <apiList.size() ; i++)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java [hunk 6]
```diff
@@ -550,6 +550,24 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
...
+            for (int i = 0; i <= apiList.size() - 1; i++) {
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java]: Any reason for not using a "for each" style loop in this case? Its a lot less simpler and less error prone

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java (hunk 5 above)
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/mappings/APIMappingUtil.java]: Any reason for not using a "for each" style loop in this case? Its a lot less simpler and less error prone

**Pull request merged**